### PR TITLE
Various Code cleanups

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
Eventough a lot of lines are changes, the changes themselves are minor.

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
I'm not sure why Travis isn't running, can someone start a build manually?

File: src/net/tcp.rs [hunk 1]
```diff
@@ -64,28 +64,28 @@ impl TcpListener {
                 match pending.poll().expect("shouldn't be canceled") {
                     Async::NotReady => {
                         self.pending_accept = Some(pending);
-                        return Err(io::ErrorKind::WouldBlock.into())
+                        return Err(io::ErrorKind::WouldBlock.into());
```
Comment (alexcrichton) [inline src/net/tcp.rs]: In general could changes like this w/ rustfmt be held off for now? It tends to be pretty burdensome to be in a "half formatted" state and it feels generally best to wait until rustfmt is ready to go and then we just turn it on all at once.

File: src/reactor/poll_evented.rs [hunk 2]
```diff
@@ -181,7 +176,7 @@ impl<E> PollEvented<E> {
     pub fn poll_ready(&self, mask: Ready) -> Async<Ready> {
         let bits = super::ready2usize(mask);
         match self.readiness.load(Ordering::SeqCst) & bits {
-            0 => {}
+            0 => {},
```
Comment (alexcrichton) [inline src/reactor/poll_evented.rs]: Could the commas here be left out? Or is this inserted by rustfmt? (if inserted by rustfmt can we hold off on that?)

File: src/reactor/poll_evented.rs [hunk 3]
```diff
@@ -275,7 +270,7 @@ impl<E> PollEvented<E> {
 impl<E: Read> Read for PollEvented<E> {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
         if let Async::NotReady = self.poll_read() {
-            return Err(io::ErrorKind::WouldBlock.into())
+            return Err(io::ErrorKind::WouldBlock.into());
```
Comment (alexcrichton) [inline src/reactor/poll_evented.rs]: Similar with the trailing commas, could stylistic changes like this be left out for now?

File: src/reactor/mod.rs [hunk 4]
```diff
@@ -166,9 +167,10 @@ impl Core {
 
         loop {
             if future_fired {
-                let res = task.poll_future_notify(&self.future_readiness, 0)?;
-                if let Async::Ready(e) = res {
-                    return Ok(e);
+                match task.poll_future_notify(&self.future_readiness, 0) {
```
Comment (alexcrichton) [inline src/reactor/mod.rs]: Could this continue to use `?` to avoid matching on `Err`?

File: src/net/udp/mod.rs [hunk 5]
```diff
@@ -376,29 +377,30 @@ impl fmt::Debug for UdpSocket {
...
 impl<T> Future for SendDgram<T>
     where T: AsRef<[u8]>,
 {
     type Item = (UdpSocket, T);
     type Error = io::Error;
 
     fn poll(&mut self) -> Poll<(UdpSocket, T), io::Error> {
-        {
-            let (ref sock, ref buf, ref addr) =
-                *self.0.as_ref().expect("SendDgram polled after completion");
```
Comment (alexcrichton) [inline src/net/udp/mod.rs]: Could this be left as-is? By using `as_ref` it avoids traffic on moves by moving the data here in and out unnecessarily.

File: src/net/udp/mod.rs [hunk 6]
```diff
@@ -415,15 +417,19 @@ impl<T> Future for RecvDgram<T>
     type Error = io::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, io::Error> {
-        let (n, addr) = {
-            let (ref socket, ref mut buf) =
-                *self.0.as_mut().expect("RecvDgram polled after completion");
-
-            try_nb!(socket.recv_from(buf.as_mut()))
+        let (sock, mut buf) = match self.0.take() {
```
Comment (alexcrichton) [inline src/net/udp/mod.rs]: Similar to the function above, could this be left as is (avoid extraneous moves in the prior version)

File: src/net/tcp.rs [hunk 7]
```diff
@@ -641,35 +641,25 @@ impl Future for TcpStreamNewState {
...
-            _ => panic!(),
+            TcpStreamNewState::Waiting(stream) => {
+                // Once we've connected, wait for the stream to be writable as
+                // that's when the actual connection has been initiated. Once
+                // we're writable we check for `take_socket_error` to see if the
+                // connect actually hit an error or not.
+                //
+                // If all that succeeded then we ship everything on up.
+                if let Async::NotReady = stream.io.poll_write() {
+                    mem::replace(self, TcpStreamNewState::Waiting(stream));
```
Comment (alexcrichton) [inline src/net/tcp.rs]: This I think can just be `*self = ...`

File: src/reactor/mod.rs [hunk 8]
```diff
@@ -365,7 +365,7 @@ fn read_ready() -> mio::Ready {
     mio::Ready::readable() | platform::hup()
 }
 
-const READ: usize = 1 << 0;
```
Comment (alexcrichton) [inline src/reactor/mod.rs]: Can this be left in? It's meant to clarify that it's just used as a bit.

**Comment by carllerche** [comment]
Thanks for splitting this up.

I think it would be best to avoid applying any style related PRs because we don't really have an established style guide and none of the style changes will be maintained. Making these style changes also clutters up the history.

That said, I think swapping `try!` -> `?` would be good to do.

**Comment by carllerche** [comment]
I'm going to close this PR though, if you want to make the `try!` -> `?` change, would you mind creating a new PR?

Thanks again!

**Comment by Thomasdezeeuw** [comment]
Is a PR with just `try!` -> `?` worth it? It might as well be done after tokio and rustfmt stabilizes and rustfmt is run.

**Comment by alexcrichton** [comment]
@Thomasdezeeuw certainly!

File: src/net/tcp.rs [hunk 9]
```diff
@@ -302,15 +298,12 @@ impl TcpStream {
     ///   loop. Note that on Windows you must `bind` a socket before it can be
     ///   connected, so if a custom `TcpBuilder` is used it should be bound
     ///   (perhaps to `INADDR_ANY`) before this method is called.
-    pub fn connect_stream(stream: net::TcpStream,
-                          addr: &SocketAddr,
-                          handle: &Handle)
-                          -> Box<Future<Item=TcpStream, Error=io::Error> + Send> {
-        let state = match mio::net::TcpStream::connect_stream(stream, addr) {
+    pub fn connect_stream(stream: net::TcpStream, addr: &SocketAddr, handle: &Handle) -> TcpStreamNew {
```
[the remark this answers is not in the document]
Comment (alexcrichton) [reply src/net/tcp.rs]: Sure!

[the remark this answers is not in the document]
Comment (Thomasdezeeuw) [reply src/net/tcp.rs]: Done https://github.com/tokio-rs/tokio/pull/66.

**Pull request closed without merging**